hswidget/dt1widget/widget.go

```diff
@@ -579,13 +580,18 @@ func (p *widget) SetTileGroup(tileGroup int32) {
 }
 
 func (p *widget) makeSubtileFlags(state *widgetState, tile *d2dt1.Tile) giu.Layout {
+	subtileFlagList := make([]string, 0)
+
+	for i := int32(0); i <= 7; i++ {
+		subtileFlagList = append(subtileFlagList, subTileString(i))
+	}
```

Comment:
I suppose, that this `7` will be pointet by `golangci-lint` as a lint error
I'd suggest to rewrite this loop:
```golang
for i := subTileFlagBlockWalk; i <= subTileFlagUnknown3;i++ {
```

Reply:
@gucio321 Just because there is a constant defined with a value of `7` somewhere doesn't mean we should just throw it in here willy nilly to appease the linter. You two can both see how it would be _utter fucking magic_ for newcomers, right?

What you're doing here is iterating over bit indices. Declaring a new constant is cheap, just declare it above the loop and it makes sense for everyone.
```golang
const bitsPerSubtileFlag = 8

for bitIdx := 0; bitIdx < bitsPerSubtileFlag; bitIdx++ {
```

Reply:
Looks good!

As far as constants go, lets turn this into a teachable moment. So the reason why 'magic numbers' are frowned upon, is because the number '8' doesn't have a specific meaning. The proper resolution is determine what that number really means, and define it as a constant -- either locally, or at a global level if its relevant to other modules. The constant isn't just to have something represent the number -- quite the opposite, the number represents the constant, thus you should only use a constant that is relevant to the condition that is being checked.